### PR TITLE
Update Gradle Dokka configuration to make sure "source" button is visible in all API docs

### PR DESCRIPTION
**Pull request by whyoleg**
For context: https://github.com/Kotlin/kotlinx.coroutines/issues/3929 and https://github.com/Kotlin/kotlinx.coroutines/pull/3960

### CODE REVIEW

File: core/build.gradle.kts [hunk 1]
```diff
@@ -381,5 +381,18 @@ tasks.withType<AbstractDokkaLeafTask>().configureEach {
...
+
+        if (name.endsWith("Main")) {
+            sourceLink {
+                // sources are located in PLATFORM/src,
+                // where PLATFORM could be jvm, js, darwin, etc
+                // configuration happens upper in the build script
+                val platform = name.dropLast(4)
+                val relPath = rootProject.projectDir.toPath().relativize(projectDir.toPath())
+                localDirectory.set(projectDir.resolve("$platform/src"))
+                remoteUrl.set(URL("https://github.com/kotlin/kotlinx-datetime/tree/master/$relPath/$platform/src"))
```
[the remark this answers is not in the document]
Comment (ilya-g) [reply core/build.gradle.kts]: It needs to be checked what sources are used to build docs for the site, but I remember we introduced the branch `latest-released` for these purposes. We advance this branch every time we make a release.

[the remark this answers is not in the document]
Comment (ilya-g) [reply core/build.gradle.kts]: I found two build configurations that build kotlinx-datetime docs, one is stable red, and the other indeed uses `latest-release` as the default branch.

[the remark this answers is not in the document]
Comment (whyoleg) [reply core/build.gradle.kts]: So, just to be sure, we need to point `remoteUrl` to `latest-release` branch ("https://github.com/kotlin/kotlinx-datetime/tree/latest-release"), as this branch will be stable until next release, while `master` branch will be used for development of new release, correct?

[the remark this answers is not in the document]
Comment (ilya-g) [reply core/build.gradle.kts]: Yes, that was the intent

[the remark this answers is not in the document]
Comment (whyoleg) [reply core/build.gradle.kts]: Nice, thanks! Pushed latest changes

**Review by ilya-g — APPROVED** [review]

**Review by dkhalanskyjb — APPROVED** [review]

**Pull request merged**